js/tabs.js

```diff
@@ -322,6 +309,54 @@ Tabs.prototype.openFiles = function() {
       this.openFileEntry.bind(this));
 };
 
+/**
+ * @param {function()} callback
+ * Invoke the save dialog for all tabs with unsaved progress. Does not close any tabs.
+ */
+Tabs.prototype.promptAllUnsaved = function(callback) {
+  this.promptAllUnsavedFromIndex_(0, callback);
```

Comment:
FMI, why not a for loop here, and then you could combine promptAllUnsavedFromIndex_ and promptAllUnsaved_ into one function?

Reply:
I thought promptAllUnsaved was nicer as a public function, happy to change if you disagree :)